src/cli/send/junitUtils.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { transformToJunit } from './junitUtils';
+
+describe('transformToJunit', () => {
+  it('should output valid request', () => {
+    const result = transformToJunit({
+      _meta: {
+        version: 'test',
+      },
+      requests: [
+        {
+          fileName: 'test.http',
+          disabled: false,
+          name: 'test',
+          duration: 1001,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+      ],
+      summary: {
+        totalRequests: 1,
+        successRequests: 1,
+        disabledRequests: 0,
+        failedRequests: 0,
+        totalTests: 1,
+        successTests: 1,
+        failedTests: 0,
+      },
+    });
+    expect(result).toBe(
+      `
+<testsuites errors="0" disabled="0" failues="0">
+  <testsuite name="test.http" tests="1" failures="0" skipped="0" package="." time="1.001" file="test.http">
+    <properties>
+      <property name="name" value="test"/>
+    </properties>
+    <testcase name="test" classname="test.http" assertions="1" time="1.001"/>
+  </testsuite>
+</testsuites>
+    `.trim()
+    );
+  });
+
+  it('should output multiple requests', () => {
+    const result = transformToJunit({
+      _meta: {
+        version: 'test',
+      },
+      requests: [
+        {
+          fileName: 'test.http',
+          disabled: false,
+          name: 'test',
+          duration: 1001,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+        {
+          fileName: 'test.http',
+          disabled: false,
+          name: 'test2',
+          duration: 1002,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+        {
+          fileName: 'test2.http',
+          disabled: false,
+          name: 'other',
+          duration: 1002,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+      ],
+      summary: {
+        totalRequests: 1,
+        successRequests: 1,
+        disabledRequests: 0,
+        failedRequests: 0,
+        totalTests: 1,
+        successTests: 1,
+        failedTests: 0,
+      },
+    });
+    expect(result).toBe(
+      `
+<testsuites errors="0" disabled="0" failues="0">
```

Comment:
if possible it would be good to include the summary information in the testsuites node attributes -- tests being the total number of tests across all child testsuite nodes. time to indicate total time it took to run all tests in all child test suites. Also having a name for the overall testsuites collection like (httpyac tests) would be useful for reporting.

Reply:
attributes added